app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java

```diff
@@ -0,0 +1,303 @@
+                {
+                    textRecognized = getString(R.string.no_text_found);
+                    final String finalTextRecognized = textRecognized;
+                    runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            mOCRTextView.setText(finalTextRecognized);
+                            tViewInci.setText(Html.fromHtml("Ingredienti trovati: "+inciDetectorEtichetta(finalTextRecognized)+""));
+                        }
+                    });
+                }
+                else
+                {
+                    final String finalTextRecognized = textRecognized;
+                    runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            mOCRTextView.setText(finalTextRecognized);
+                            tViewInci.setText(Html.fromHtml("Ingredienti trovati: "+inciDetectorEtichetta(finalTextRecognized)+""));
+                        }
+                    });
+                }
+            } else {
+                Log.e("NOT_FOUND", "photo not found");
+            }
+            return textRecognized;
+        }
+
+        @Override
+        protected void onPostExecute(String s) {
+            progressDialog.dismiss();
+            // Saving in the preferences
+            SharedPreferences sharedPref = getApplicationContext().getSharedPreferences("prefs", Context.MODE_PRIVATE);
+            SharedPreferences.Editor editor = sharedPref.edit();
+            editor.putString("text", s);
+            editor.apply();
+        }
+
+        @Override
+        protected void onPreExecute() {
+            progressDialog = ProgressDialog.show(ResultActivity.this,
+                    progressMessage,
+                    "");
+        }
+    }
+
+
+    public Bitmap rotateBitmap(Bitmap original, float degrees) {
+        int width = original.getWidth();
+        int height = original.getHeight();
+
+        Matrix matrix = new Matrix();
+        matrix.preRotate(degrees);
+
+        Bitmap rotatedBitmap = Bitmap.createBitmap(original, 0, 0, width, height, matrix, true);
+        Bitmap mutableBitmap = rotatedBitmap.copy(Bitmap.Config.ARGB_8888, true);
+        Canvas canvas = new Canvas(mutableBitmap);
+        canvas.drawBitmap(original, 5.0f, 0.0f, null);
+
+        return rotatedBitmap;
+    }
+
+    /**
+     * @author Giovanni Fasan(g1), Giovanni Piva(g1)
+     * @param text String in which you have to find the ingredients
+     * @return String with the list of ingredients found in the text
+     * Put in database csv: https://www.youtube.com/watch?v=i-TqNzUryn8
+     */
+    public String inciDetector(String text){
+        String inci = "";
+        try {
+            InputStream is = getResources().openRawResource(R.raw.database);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(is, Charset.forName("UTF-8")));
```

Comment:
i didn't understand what is "UTF-8". 
If it's a name could be better to define it as a final variable to avoid HardCoding.

The same problem in line 273